client/packages/common/src/ui/forms/JsonForms/useJsonForms.tsx

```diff
@@ -77,14 +113,51 @@ export const useJsonForms = (docName: string) => {
     { tester: arrayTester, renderer: Array },
   ];
 
+  const showSaveConfirmation = useConfirmationModal({
+    onConfirm: saveData,
+    message: t('messages.confirm-save-generic'),
+    title: t('heading.are-you-sure'),
+  });
+
+  const showCancelConfirmation = useConfirmationModal({
+    onConfirm: onCancel,
+    message: t('messages.confirm-cancel-generic'),
+    title: t('heading.are-you-sure'),
```

Comment:
I wonder about passing in the key used here - the current message is safely generic, but maybe the user would like to show something specific

Reply:
Yeah, in light of this and the last few comments, I've made all the generic messages customisable with props options. 

```
interface JsonFormOptions {
  showButtonPanel?: boolean;
  onCancel?: () => void;
  saveConfirmationMessage?: string;
  cancelConfirmationMessage?: string;
  saveSuccessMessage?: string;
}
```

97c6aabe8